androidbrowserhelper/src/test/java/com/google/androidbrowserhelper/trusted/ManageDataLauncherActivityTest.java

```diff
@@ -0,0 +1,203 @@
+        mShadowPackageManager = shadowOf(mPackageManager);
+    }
+
+    @Test
+    public void chrome72SupportsSiteSettings() {
+        installChrome(CHROME_72_VERSION);
+        TwaProviderPicker.Action action = TwaProviderPicker.pickProvider(mPackageManager);
+
+        assertTrue(ManageDataLauncherActivity.packageSupportsSiteSettings(action.provider, mPackageManager));
+    }
+
+    @Test
+    public void chrome71DoesNotSupportsSiteSettings() {
+        installChrome(CHROME_71_VERSION);
+        TwaProviderPicker.Action action = TwaProviderPicker.pickProvider(mPackageManager);
+
+        assertFalse(ManageDataLauncherActivity.packageSupportsSiteSettings(action.provider, mPackageManager));
+    }
+
+    @Test
+    public void browserWithCategorySupportsSiteSettings() {
+        installTrustedWebActivityProviderWithSiteSettingsCategory(TWA_PROVIDER_PACKAGE);
+
+        TwaProviderPicker.Action action = TwaProviderPicker.pickProvider(mPackageManager);
+
+        assertTrue(ManageDataLauncherActivity.packageSupportsSiteSettings(action.provider, mPackageManager));
+    }
+
+    @Test
+    public void browserWithoutCategorySupportsSiteSettings() {
+        installTrustedWebActivityProvider(TWA_PROVIDER_PACKAGE);
+
+        TwaProviderPicker.Action action = TwaProviderPicker.pickProvider(mPackageManager);
+
+        assertFalse(ManageDataLauncherActivity.packageSupportsSiteSettings(action.provider, mPackageManager));
+    }
+
+    @Test
+    @Config(sdk = Build.VERSION_CODES.N_MR1)
+    public void returnsSiteSettingsShortcut() {
+        installTrustedWebActivityProviderWithManageDataAction(TWA_PROVIDER_PACKAGE);
+
+        assertNotNull(ManageDataLauncherActivity.getSiteSettingsShortcutOrNull(mContext, mPackageManager));
+    }
+
+    @Test
+    @Config(sdk = Build.VERSION_CODES.N)
+    public void returnsNullSiteSettingsShortcutWithLowSDK() {
+        installTrustedWebActivityProviderWithManageDataAction(TWA_PROVIDER_PACKAGE);
+
+        assertNull(ManageDataLauncherActivity.getSiteSettingsShortcutOrNull(mContext, mPackageManager));
+    }
+
+    @Test
+    @Config(sdk = Build.VERSION_CODES.N_MR1)
+    public void returnsNullSiteSettingsShortcutWhenMissingManageDataLauncherActivity() {
+        assertNull(ManageDataLauncherActivity.getSiteSettingsShortcutOrNull(mContext, mPackageManager));
+    }
+
+    private void installBrowser(String packageName) {
+        Intent intent = new Intent()
+                .setData(Uri.fromParts("http", "", null))
+                .setAction(Intent.ACTION_VIEW)
+                .addCategory(Intent.CATEGORY_BROWSABLE);
+
+        ResolveInfo resolveInfo = new ResolveInfo();
+        resolveInfo.activityInfo = new ActivityInfo();
+        resolveInfo.activityInfo.packageName = packageName;
+
+        mShadowPackageManager.addResolveInfoForIntent(intent, resolveInfo);
+    }
+
+    private void installCustomTabsProvider(String packageName) {
```

Comment:
Maybe it's worth sharing code between this and `TwaProviderPickerTest` - it looks like you've just copied it without any changes.